components/merkledb/src/indexes/proof_map/node.rs

```diff
@@ -27,6 +27,9 @@ use crate::{BinaryKey, BinaryValue, HashTag, ObjectHash};
 const BRANCH_NODE_SIZE: usize = 2 * (HASH_SIZE + PROOF_PATH_SIZE);
 
 /// Enum that represents the type of the `ProofMap` node.
+///
+/// This type is not intended to be exhaustively matched. It can be extended in the future
```

Comment:
This type is not public, is it? If it is, I'd view publicity as a problem, not exhaustive matching.

Reply:
Ah, I automated the process of adding non-exhaustive guards, but when I was sifting them, forgot to remove comments in several places.